SH4: simde/x86/sse.h:2906:10: error: 'FE_UPWARD' was not declared in this scope

Full log at https://buildd.debian.org/status/fetch.php?pkg=libssw&arch=sh4&ver=1.1-6%7E0expsimde0&stamp=1576178932&raw=0

```
g++ -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -DSIMDE_ENABLE_OPENMP -fopenmp-simd -O3 -DSIMDE_ENABLE_OPENMP -fopenmp-simd -O3 -fPIC -shared -rdynamic -Wl,-soname,libssw.so.0 -o libssw.so.0 ssw.c ssw.h ssw_cpp.h ssw_cpp.cpp -Wl,-z,relro -Wl,-z,now
In file included from ../debian/include/simde/x86/sse2.h:34,
                 from ssw.c:38:
../debian/include/simde/x86/sse.h: In function 'uint32_t simde_mm_getcsr()':
../debian/include/simde/x86/sse.h:2906:10: error: 'FE_UPWARD' was not declared in this scope
 2906 |     case FE_UPWARD:
      |          ^~~~~~~~~
../debian/include/simde/x86/sse.h:2909:10: error: 'FE_DOWNWARD' was not declared in this scope; did you mean 'FP_INT_DOWNWARD'?
 2909 |     case FE_DOWNWARD:
      |          ^~~~~~~~~~~
      |          FP_INT_DOWNWARD
../debian/include/simde/x86/sse.h: In function 'void simde_mm_setcsr(uint32_t)':
../debian/include/simde/x86/sse.h:2935:18: error: 'FE_DOWNWARD' was not declared in this scope; did you mean 'FP_INT_DOWNWARD'?
 2935 |       fesetround(FE_DOWNWARD);
      |                  ^~~~~~~~~~~
      |                  FP_INT_DOWNWARD
../debian/include/simde/x86/sse.h:2938:18: error: 'FE_UPWARD' was not declared in this scope
 2938 |       fesetround(FE_UPWARD);
      |                  ^~~~~~~~~
```

@nemequ closer!

```
g++ -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -DSIMDE_ENABLE_OPENMP -fopenmp-simd -O3 -DSIMDE_ENABLE_OPENMP -fopenmp-simd -O3 -fPIC -shared -rdynamic -Wl,-soname,libssw.so.0 -o libssw.so.0 ssw.c ssw.h ssw_cpp.h ssw_cpp.cpp -Wl,-z,relro -Wl,-z,now
In file included from ../debian/include/simde/x86/sse2.h:34,
                 from ssw.c:38:
../debian/include/simde/x86/sse.h: In function ‘simde__m128 simde_mm_cmpord_ps(simde__m128, simde__m128)’:
../debian/include/simde/x86/sse.h:771:17: error: ‘isnan’ was not declared in this scope; did you mean ‘std::isnan’?
  771 |     r.u32[i] = (isnan(a.f32[i]) || isnan(b.f32[i])) ? 0 : 0xffffffff;
      |                 ^~~~~
      |                 std::isnan
```

--- a/simde/x86/sse.h
+++ b/simde/x86/sse.h
@@ -68,13 +68,8 @@
 #    endif
 #  endif
 
-#if !defined(__cplusplus)
-#  include <math.h>
-#  include <fenv.h>
-#else
-#  include <cmath>
-#  include <cfenv>
-#endif
+#include <math.h>
+#include <fenv.h>
 
 HEDLEY_DIAGNOSTIC_PUSH
 #  if HEDLEY_HAS_WARNING("-Wfloat-equal")

--- a/simde/x86/sse.h
+++ b/simde/x86/sse.h
@@ -68,8 +68,13 @@
 #    endif
 #  endif
 
+#if !defined(__cplusplus)
 #  include <math.h>
 #  include <fenv.h>
+#else
+#  include <cmath>
+#  include <cfenv>
+#endif
 
 HEDLEY_DIAGNOSTIC_PUSH
 #  if HEDLEY_HAS_WARNING("-Wfloat-equal")
@@ -2901,17 +2906,25 @@ simde_mm_getcsr (void) {
   int rounding_mode = fegetround();
 
   switch(rounding_mode) {
+#if defined(FE_TONEAREST)
     case FE_TONEAREST:
       break;
+#endif
+#if defined(FE_UPWARD)
     case FE_UPWARD:
       r |= 2 << 13;
       break;
+#endif
+#if defined(FE_DOWNWARD)
     case FE_DOWNWARD:
       r |= 1 << 13;
       break;
+#endif
+#if defined(FE_TOWARDZERO)
     case FE_TOWARDZERO:
       r = 3 << 13;
       break;
+#endif
   }
 
   return r;
@@ -2928,18 +2941,26 @@ simde_mm_setcsr (uint32_t a) {
   _mm_setcsr(a);
 #else
   switch((a >> 13) & 3) {
+#if defined(FE_TONEAREST)
     case 0:
       fesetround(FE_TONEAREST);
+#endif
+#if defined(FE_DOWNWARD)
       break;
     case 1:
       fesetround(FE_DOWNWARD);
+#endif
+#if defined(FE_UPWARD)
       break;
     case 2:
       fesetround(FE_UPWARD);
+#endif
+#if defined(FE_TOWARDZERO)
       break;
     case 3:
       fesetround(FE_TOWARDZERO);
       break;
+#endif
   }
 #endif
 }

